src/client/Client.cc

```diff
@@ -2552,7 +2552,7 @@ void Client::handle_client_request_forward(const MConstRef<MClientRequestForward
    * using the hardcode here.
    */
   int max_fwd = sizeof(((struct ceph_mds_request_head*)0)->num_fwd);
-  max_fwd = 1 << (max_fwd * CHAR_BIT) - 1;
+  max_fwd = (1 << (max_fwd * CHAR_BIT)) - 1;
```

Comment:
The expression `1 << (max_fwd * CHAR_BIT) - 1` is same as `1 << ((max_fwd * CHAR_BIT) - 1)` and not same as `(1 << (max_fwd * CHAR_BIT)) - 1`. After discussion with Xiubo, I found that the desired value is 255. So the first two are incorrect and last one is correct.

I've created a tracker ticket for this bug. @lxbsz Can you please include it? Here's the link - https://tracker.ceph.com/issues/55409

Reply:
Thanks @rishabh-d-dave very much for this. I have updated it.

Reply:
@joscollin Update this PR and nothing changed in the code, just add more in the commit comments and add the tracker link to the commit.

Reply:
> And also the '_' has precedence over the '<<', more detail please

It's '-'

Reply:
Done.

Reply:
Looks good, already approved.